packages/api-format/src/codec/String.ts

```diff
@@ -0,0 +1,58 @@
+// Copyright 2017-2018 @polkadot/api-format authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import { Base } from './types';
+
+import u8aFromUtf8 from '@polkadot/util/u8a/fromUtf8';
+import u8aToUtf8 from '@polkadot/util/u8a/toUtf8';
+import u8aConcat from '@polkadot/util/u8a/concat';
+
+// NOTE or LengthCompact
+import Length from './base/Length';
+
+export default class String implements Base<string> {
+  private _length: Length;
+
+  raw: string;
+
+  constructor (value: string = '') {
+    this._length = new Length(value.length);
+    this.raw = value;
+  }
+
+  byteLength (): number {
```

Comment:
As with Arrays, we don't have length, in any of the classes. (Well, we do have in String and Array). `byteLength` has the total length including the length of the prefixes, where length is just the number of elements. (So in coding use byteLength, when working with data, just a straight length is fine)